powershell/README.md

```diff
@@ -81,7 +81,149 @@ Then, you can install a pre-release version of Safeguard-DevOps by running:
+up to the point of configuring the Safeguard A2A registration and registering accounts
+whose credentials should be mapped to various third party vault plugins. However, at one
+point during the initial configuration, it will prompt for a client certificate. If your
+company has a client certificate including a private key that can be uploaded to the
+Secrets Broker, the cmdlet will allow you to complete the configuration in one step. Otherwise
+additional commandlets will need to be invoked.  The following is a common configuration
+scenario if a new client certificate needs to be generated by the Secrets Broker
+
+The cmdlet will provide additional instructions and prompts at this point.
+
+```Powershell
+> Initialize-SgDevOps
+Appliance: <IP address or DNS name of your Safeguard Appliance>
+ServiceAddress: <Ip Address or DNS name of your Secrets Broker>
+
+** Safeguard and Secrets Broker TLS certificate that should be validated. Follow prompts.
+
+** Prompt to login
+(certificate, local)
+Provider: local
+Username: <Local Safeguard user>
+Password: *******
+** The Secrets Broker is associated with a Safeguard Appliance
+
+** Login to the Secrets Broker to Complete the configuration
+(certificate, local)
+Provider: local
+Username: <Local Safeguard User>
+Password: *******
+```
+
+The Secrets Broker prompts for a client certificate which will need to be provided later.
+Answer 'N' to the prompt. This will end the initial configuration. The rest of the
+configuration will be done using individual cmdlets.
+
+```Powershell
+> Connect-SgDevOps -Insecure <IP address or DNS of the Secrets Broker>
+(certificate, local)
+Provider: local
+Username: <Local Safeguard User>
+Password: *******
+```
+
+The first requirement is to create the client certificate.
+
+```Powershell
+> New-SgDevOpsCsr
+CertificateType: client
+Subject: cn=<your subject>...
+```
+
+The Secrets Broker will display the generated CSR as well as save the CSR to a file
+called 'client.csr' in the current directory. The CSR will need to be signed using your
+Company's PKI.
+
+```Powershell
+> Install-SgDevOpsClientCertificate
+CertificateFile: <Full path to the signed certificate>
+For no password just press enter...
+Password:
+```
+
+The new client certificate is uploaded and installed into the Secrets Broker
+
+This cmdlet finishes the Secrets Broker connection configuration with the Safeguard
+appliance by creating a A2A certificate user and A2A registration in Safeguard.
+
+```Powershell
+> Initialize-SgDevOpsConfiguration
+```
+
+Find and register available Safeguard accounts with the Secrets Broker A2A registration.
+Use Find-SgDevOpsAvailableAssetAccount or Get-SgDevOpsAvailableAssetAccount to discover
+the account that should be registered with the Secrets Broker. See the documentation for
+each of these cmdlets for filtering and searching options.
+
+```Powershell
+> Get-SgDevOpsAvailableAssetAccount
```

Comment:
You should use the setup cmdlet here that searches for available assets.  If you run this cmdlet on a reasonably sized customer data source it is unusable.  It could return 50,000 results.

Reply:
Invoke-SgDevOpsRegisteredAccountSetup

Reply:
@bnicholesdell